core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -128,18 +129,27 @@ public List<Application<?>> getApplications() {
     return Collections.unmodifiableList(applications);
   }
 
-  private void attachMetricsListener(Metrics metrics, Map<String, String> tags) {
-    MetricsListener metricsListener = new MetricsListener(metrics, "jetty", tags);
+  private void attachMetricsListener(String listenerName, Metrics metrics,
+      Map<String, String> tags) {
     for (NetworkTrafficServerConnector connector : connectors) {
-      connector.addNetworkTrafficListener(metricsListener);
+      if (Objects.equals(connector.getName(), listenerName)) {
```

Comment:
My understanding is that there would be atleast 1 listener per app, can you throw if its not the case?

Reply:
> My understanding is that there would be atleast 1 listener per app, can you throw if its not the case?

I meant each app would have atleast 1 corresponding network-connector, and this connector would get metric-listener registered on it. So assert that in `attachMetricsListener` there is atleast 1 network-connector for the input app. Just some defensive programming to avoid future errors.